### ISSUE TITLE
Bug: csv file lost bom when download more than one time.

### ISSUE THREAD

**Issue by Lumuy** [activeadmin/activeadmin#7036]
### Did you find a bug?

When download csv file on index page the second time, the `bom` will lost.

If the csv file contain Chinese or Japanese, it act normal on Mac system opening with Excel, but on Windows system opening with Excel, can not see the right content.

### Expected behavior

When download csv file lots of time, opening with Excel on Windows system, see the right encoding context.

### Actual behavior

Csv file lost bom.

### How to reproduce

On index page, download csv file more than one time, except the first time, others which contain Japanese or Chinese won't show the right format.

### Reason
      
https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/csv_builder.rb#L48
` bom = options.delete :byte_order_mark `
The second time you download csv file, `options` already deleted :byte_order_mark since the first time, so bom will be nil.
I try to make a PR, but there's another problem #7032

**Comment by Karoid** [comment]
I've made PR for this bug, but it seems to take time for fixing problem.
You can temporarily monkey patch this problem by adding following line at the end of `config/initializers/active_admin.rb`
```ruby
class ActiveAdmin::CSVBuilder
  def build(controller, csv)
    @collection = controller.send :find_collection, except: :pagination
    columns = exec_columns controller.view_context
    bom = options[:byte_order_mark]
    column_names = options.delete(:column_names) { true }
    csv_options = options.except :encoding_options, :humanize_name, :byte_order_mark

    csv << bom if bom

    if column_names
      csv << CSV.generate_line(columns.map { |c| encode c.name, options }, **csv_options)
    end

    ActiveRecord::Base.uncached do
      (1..paginated_collection.total_pages).each do |page|
        paginated_collection(page).each do |resource|
          resource = controller.send :apply_decorator, resource
          csv << CSV.generate_line(build_row(resource, columns, options), **csv_options)
        end
      end
    end

    csv
  end
end
```

**Comment by javierjulio** [comment]
@Karoid would your PR #7170 resolve both this issue and #7032 that was mentioned in this issue's description? Just wanted to confirm.

**Comment by Karoid** [comment]
@javierjulio I checked for #7032 again and found there was some problem in my PR(#7170)
So I've fixed the problem and pushed again(#7170)
